app/services/decisions/custom_wrapper.rb

```diff
@@ -3,5 +3,12 @@ class CustomWrapper < SimpleDelegator
     def any_invalid?(scope, *forms)
       scope.detect { |record| forms.any? { |f| !f.build(record, application:).valid? } }
     end
+
+    def source_was_check_answers?(rule, any_source: false)
```

Comment:
So I think this approach can still work, but we need to find a way of making it feel less like a hack.

I think instead of the DecisionTree saying "after screen X go to the start page" and then us hijacking it to go to the check answers page instead, the DecisionTree should say "after screen X go to whatever the appropriate summary page is", and then either here or elsewhere, our logic should run to say "the' appropriate summary page' is either the start page or the check answers page depending on the state history.